osu.Game.Rulesets.Osu/UI/OsuTouchInputMapper.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Input;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+
+namespace osu.Game.Rulesets.Osu.UI
+{
+    public class OsuTouchInputMapper : Drawable
+    {
+        /// <summary>
+        /// The default cursor touch it may fail <see cref="IsCursorTouch(TouchSource)"/> on cases where there aren't any cursor touches, such as when autopilot is on.
```

Comment:
I have no idea what this is trying to say. It feels like three sentences mashed into one, all touching on different points.

Reply:
I will try to document it better when i get home, sorry.

Reply:
I believe i explained better what the `DEFAULT_CURSOR_TOUCH` means on my recent commits.